polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -98,3 +116,873 @@ impl<BlockNumber: Default + From<u32>> Default for SchedulerParams<BlockNumber>
+				super::v7::CandidateEvent::CandidateIncluded(
+					receipt.into(),
+					head_data,
+					core_index,
+					group_index,
+				),
+			CandidateEvent::CandidateTimedOut(receipt, head_data, core_index) =>
+				super::v7::CandidateEvent::CandidateTimedOut(receipt.into(), head_data, core_index),
+		}
+	}
+}
+
+impl<H> CandidateReceiptV2<H> {
+	/// Get a reference to the candidate descriptor.
+	pub fn descriptor(&self) -> &CandidateDescriptorV2<H> {
+		&self.descriptor
+	}
+
+	/// Computes the blake2-256 hash of the receipt.
+	pub fn hash(&self) -> CandidateHash
+	where
+		H: Encode,
+	{
+		CandidateHash(BlakeTwo256::hash_of(self))
+	}
+}
+
+impl<H: Clone> CommittedCandidateReceiptV2<H> {
+	/// Transforms this into a plain `CandidateReceipt`.
+	pub fn to_plain(&self) -> CandidateReceiptV2<H> {
+		CandidateReceiptV2 {
+			descriptor: self.descriptor.clone(),
+			commitments_hash: self.commitments.hash(),
+		}
+	}
+
+	/// Computes the hash of the committed candidate receipt.
+	///
+	/// This computes the canonical hash, not the hash of the directly encoded data.
+	/// Thus this is a shortcut for `candidate.to_plain().hash()`.
+	pub fn hash(&self) -> CandidateHash
+	where
+		H: Encode,
+	{
+		self.to_plain().hash()
+	}
+
+	/// Does this committed candidate receipt corresponds to the given [`CandidateReceipt`]?
+	pub fn corresponds_to(&self, receipt: &CandidateReceiptV2<H>) -> bool
+	where
+		H: PartialEq,
+	{
+		receipt.descriptor == self.descriptor && receipt.commitments_hash == self.commitments.hash()
+	}
+}
+
+impl PartialOrd for CommittedCandidateReceiptV2 {
+	fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+		Some(self.cmp(other))
+	}
+}
+
+impl Ord for CommittedCandidateReceiptV2 {
+	fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+		self.descriptor
+			.para_id
+			.cmp(&other.descriptor.para_id)
+			.then_with(|| self.commitments.head_data.cmp(&other.commitments.head_data))
+	}
+}
+
+impl<H: Copy> From<CommittedCandidateReceiptV2<H>> for super::v7::CommittedCandidateReceipt<H> {
+	fn from(value: CommittedCandidateReceiptV2<H>) -> Self {
+		Self { descriptor: value.descriptor.into(), commitments: value.commitments }
+	}
+}
+
+impl<H: Copy> From<CandidateReceiptV2<H>> for super::v7::CandidateReceipt<H> {
+	fn from(value: CandidateReceiptV2<H>) -> Self {
+		Self { descriptor: value.descriptor.into(), commitments_hash: value.commitments_hash }
+	}
+}
+
+/// A strictly increasing sequence number, tipically this would be the least significant byte of the
+/// block number.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, TypeInfo, RuntimeDebug)]
+pub struct CoreSelector(pub u8);
+
+/// An offset in the relay chain claim queue.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, TypeInfo, RuntimeDebug)]
+pub struct ClaimQueueOffset(pub u8);
+
+/// Default claim queue offset
+pub const DEFAULT_CLAIM_QUEUE_OFFSET: ClaimQueueOffset = ClaimQueueOffset(1);
```

Comment:
I think it would make more sense for this to be 0 (building on the latest relay parent), as this is what all collators do for now anyway

Reply:
Setting it to 0 would pretty much make it sync backing, as the candidate would need to be backed in the very next relay chain block. With 1 you have more time to build the collation and fill the block.

Reply:
anyway I don't think this default value is going to be used on the relay chain, right?

but it won't be sync backing, because you can back canddiates built on the previous relay chain block. Both the lookahead collator and the slot-based collator currently build based on the first item in the claimqueue

Reply:
Yeah, the slot based collator needs to be updated to build exactly on the core indicated by the core selector.

Reply:
Indeed it doesn't make sense on the relay chain, so I would just move it and use in tests only here.